examples/tree.rs

```diff
@@ -0,0 +1,61 @@
+extern crate turtle;
+
+use turtle::*;
+
+fn draw(turtle: &mut Turtle, side: f64, deep: usize) {
+    let next_side = side / f64::sqrt(2.0);
+
+    for _ in 0..4 {
+        let position = turtle.position();
+        turtle.set_pen_color(get_color(position));
+        turtle.right(90.0);
+        turtle.forward(side);
+    }
+
+    if deep > 0 {
+        turtle.pen_up();
+        turtle.left(45.0);
+        turtle.forward(next_side);
+        let position = turtle.position();
+        turtle.pen_down();
+
+        let heading = turtle.heading();
+
+        draw(turtle, next_side, deep - 1);
+        turtle.set_heading(heading);
+
+        turtle.pen_up();
+        turtle.go_to(position);
+        turtle.right(135.0);
+        turtle.forward(side);
+        turtle.left(45.0);
+        turtle.forward(next_side);
+        turtle.pen_down();
+
+        let heading = turtle.heading();
+
+        draw(turtle, next_side, deep - 1);
+        turtle.set_heading(heading);
+    }
+}
+
+fn get_color(position: [f64;2]) -> Color {
```

Comment:
If you run this example with `cargo run --example tree`, there is a [compile error](https://travis-ci.org/sunjay/turtle/jobs/327270457#L1224). This is caused by a slight breaking change in turtle itself (our fault, not yours). To fix it, just change `[f64; 2]` to `Point`. Everything else should be the same.